lightning/src/chain/mod.rs

```diff
@@ -69,7 +69,7 @@ pub trait Access: Send + Sync {
 /// [`PermanentFailure`]: channelmonitor/enum.ChannelMonitorUpdateErr.html#variant.PermanentFailure
 pub trait Watch: Send + Sync {
 	/// Keys needed by monitors for creating and signing transactions.
-	type Keys: ChannelKeys;
+	type ChanSigner: ChannelSigner;
```

Comment:
Not sure if you're warming to the using verbs for traits and nouns for types, but consider `trait Sign` and `type ChannelSigner: Sign` for consistency. Then you could dispense with abbreviated `ChanSigner`.

Reply:
Considering the options in more detail:

 N | Trait | Type | Type Param
--- | --- | --- | ---
1 | ChannelSigner | InMemoryChannelSigner | ChanSigner
2 | Sign | InMemoryChannelSigner | ChannelSigner
3 | ChannelSign | InMemoryChannelSigner | ChannelSigner
4 | Sign | InMemorySigner | Signer

So 1 is this PR in its current state, 2 is what @jkczyz proposed.  3 seems to be a more consistent version of 2 which doesn't drop the word `Channel` on the trait.  We could go for 4 given that signing at the channel level is such a fundamental aspect of a Lightning node, so might as well be a bit less verbose.

I prefer 4 somewhat, but I don't feel strongly.

Reply:
Hmm, the problem I have with Sign is it doesn't really fully capture what's going on - it's a channel-specific thing that signs. If we moved it to channel::Sign it'd be more like chain::Watch but I'm not sure it makes sense there.

Reply:
I tend to agree with @devrandom in that "channel" is fundamental so could be dropped. I'd favor (4) but with slight variation.

IMO, it is redundant when used as the type parameter to `Channel`.

```rust
pub(super) struct Channel<ChanSigner: ChannelSigner> {
}

impl<ChanSigner: ChannelSigner> Channel<ChanSigner> {
}
```

That it is for a specific channel could be made clear in the documentation and is enforced by it being a type parameter for `Channel` and `ChannelMonitor`.

However, for parameterizing types where what signing pertains to isn't clear, the type parameter could be the more specific `ChannelSigner` instead `Signer` (e.g., in `Watch`).

Reply:
Ok, I think I did this everywhere, let me know if I missed any cases.